dotnet/packages/Microsoft.TeamsAI/Microsoft.TeamsAI/AI/Moderator/AzureContentSafetyModerator.cs

```diff
@@ -86,41 +78,57 @@ public async Task<Plan> ReviewPlan(ITurnContext turnContext, TState turnState, P
-            AzureContentSafetyTextAnalysisRequest request = new(text)
+            AnalyzeTextOptions analyzeTextOptions = new(text);
+            if (_options.Categories != null)
             {
-                BlocklistNames = _options.BlocklistNames,
-                Categories = _options.Categories,
-                BreakByBlocklists = _options.BreakByBlocklists,
-            };
-
-            AzureContentSafetyTextAnalysisResponse response = await _client.ExecuteTextModeration(request);
-
-            bool flagged = response.BlocklistsMatchResults.Count > 0
-                || _ShouldBeFlagged(response.HateResult)
-                || _ShouldBeFlagged(response.SelfHarmResult)
-                || _ShouldBeFlagged(response.SexualResult)
-                || _ShouldBeFlagged(response.ViolenceResult);
-            if (flagged)
+                foreach (TextCategory category in _options.Categories)
+                {
+                    analyzeTextOptions.Categories.Add(category);
+                }
+            }
+            if (_options.BlocklistNames != null)
             {
-                string actionName = isModelInput ? DefaultActionTypes.FlaggedInputActionName : DefaultActionTypes.FlaggedOutputActionName;
+                foreach (string blocklistName in _options.BlocklistNames)
+                {
+                    analyzeTextOptions.BlocklistNames.Add(blocklistName);
+                }
+            }
 
-                // Flagged
-                return new Plan()
+            try
+            {
+                Response<AnalyzeTextResult> response = await _client.AnalyzeTextAsync(analyzeTextOptions);
+
+                bool flagged = response.Value.BlocklistsMatchResults.Count > 0
+                || _ShouldBeFlagged(response.Value.HateResult)
+                || _ShouldBeFlagged(response.Value.SelfHarmResult)
+                || _ShouldBeFlagged(response.Value.SexualResult)
+                || _ShouldBeFlagged(response.Value.ViolenceResult);
+                if (flagged)
                 {
-                    Commands = new List<IPredictedCommand>
+                    string actionName = isModelInput ? DefaultActionTypes.FlaggedInputActionName : DefaultActionTypes.FlaggedOutputActionName;
+
+                    // Flagged
+                    return new Plan()
+                    {
+                        Commands = new List<IPredictedCommand>
                             {
                                 new PredictedDoCommand(actionName, new Dictionary<string, object>
                                 {
-                                    { "Result", response }
+                                    { "Result", response.Value }
                                 })
                             }
-                };
-            }
+                    };
+                }
 
-            return null;
+                return null;
+            }
+            catch (RequestFailedException)
+            {
+                throw;
```

Comment:
Here seems just catch and throw, then how about removing the try-catch and let exception to directly throw?

Reply:
Nice idea, removed try-catch.